### PR TITLE
353 fix/delete post revisions test review

### LINKED ISSUE #353: Delete Post Revisions: Add Tests
Blocked by #178

### PR DESCRIPTION
**Pull request by rajanit2000**
@mariadanieldeepak its ready to review 

Blocked by #332 

Fix #353

### CODE REVIEW

File: include/Core/Base/Mixin/Renderer.php [hunk 1]
```diff
@@ -111,4 +111,54 @@ protected function render_user_role_dropdown() {
...
 		bd_render_post_type_dropdown( $this->field_slug );
 	}
+
+	/**
+	 * Render Taxonomy dropdown.
+	 */
+	protected function render_taxonomy_dropdown() {
+		$taxonomies = get_taxonomies( array(), 'objects' );
+		?>
+		<select name="smbd_<?php echo esc_attr( $this->field_slug ); ?>_taxonomy" class="select2" multiple="multiple" data-placeholder="<?php _e( 'Select Taxonomy', 'bulk-delete' ); ?>">
```
Comment (mariadanieldeepak) [inline include/Core/Base/Mixin/Renderer.php]: Avoid using `select2` in class names. Know more on why you should avoid it [here](https://github.com/sudar/bulk-delete/pull/365#issue-199372401).

Comment (mariadanieldeepak) [inline include/Core/Base/Mixin/Renderer.php]: Avoid using `select2` in class names. Know more on why you should avoid it [here](https://github.com/sudar/bulk-delete/pull/365#issue-199372401).

**Comment by mariadanieldeepak** [comment]
@rajanit2000,

Before sending PR kindly make sure you fix all Codacy issues.

File: include/Core/Base/Mixin/Renderer.php [hunk 2]
```diff
@@ -111,4 +111,54 @@ protected function render_user_role_dropdown() {
...
+	}
+
+	/**
+	 * Render term options.
+	 */
+	protected function render_term_options() {
+		?>
+		<h4><?php _e( 'Delete terms that', 'bulk-delete' ); ?></h4>
+		<select name="smbd_<?php echo esc_attr( $this->field_slug ); ?>_term_opt">
+		<?php if( $this->field_slug == 'terms_by_name' ){ ?>
```
Comment (mariadanieldeepak) [inline include/Core/Base/Mixin/Renderer.php]: Incorrect formatting used in `if` condition.

File: include/Core/Base/Mixin/Renderer.php [hunk 3]
```diff
@@ -111,4 +111,54 @@ protected function render_user_role_dropdown() {
...
+			<option value="equal_to">equal to</option>
+			<option value="not_equal_to">not equal to</option>
+			<option value="starts">starts</option>
+			<option value="ends">ends</option>
+			<option value="contains">contains</option>
+			<option value="non_contains">non contains</option>
+		<?php }elseif( $this->field_slug == 'terms_by_post_count' ){ ?>
```
Comment (mariadanieldeepak) [inline include/Core/Base/Mixin/Renderer.php]: In correct formatting used in `else` statement.

File: include/Core/BulkDelete.php [hunk 4]
```diff
@@ -467,6 +472,22 @@ private function get_delete_metas_admin_page() {
 		return $metas_page;
 	}
 
+	/**
+	 * Get Bulk Delete Terms admin page.
```
Comment (mariadanieldeepak) [inline include/Core/BulkDelete.php]: The Docblock comment should say what the function does and not what the should do. For example, "Get**s** Bulk Delete Term's admin page" is correct.

File: include/Core/BulkDelete.php [hunk 5]
```diff
@@ -467,6 +472,22 @@ private function get_delete_metas_admin_page() {
...
+	 *
+	 * @since 6.0.0
+	 *
+	 * @return \BulkWP\BulkDelete\Core\Terms\DeleteTermsPage
+	 */
+	private function get_delete_terms_admin_page() {
```
Comment (mariadanieldeepak) [inline include/Core/BulkDelete.php]: Private functions should use `@access private` tag in the Docblock as per [WordPress inline documentation](https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/php/) standards.

**Comment by rajanit2000** [comment]
Its just like #332
Codacy still is not passed, But want to review
Codecy issue something we want to ignore it
https://app.codacy.com/app/rajanit2000/bulk-delete/pullRequest?prid=1904421

**Comment by rajanit2000** [comment]
Development is done, Waiting for review

File: include/Core/BulkDelete.php (hunk 5 above)
Comment (rajanit2000) [reply include/Core/BulkDelete.php]: @mariadanieldeepak, @sudar says ignore the change

**Comment by rajanit2000** [comment]
@sudar Its ready to review

File: tests/wp-unit/include/Core/Posts/Modules/DeletePostsByRevisionModuleTest.php [hunk 6]
```diff
@@ -20,20 +20,60 @@ class DeletePostsByRevisionModuleTest extends WPCoreUnitTestCase {
...
+			'post_content' => md5( rand() ),
+		);
+
+		wp_update_post( $revision_post_2 );
+
+		$delete_options = array(
+			'revisions' => 'revisions',
+		);
+
+		$posts_deleted = $this->module->delete( $delete_options );
```
[the remark this answers is not in the document]
Comment (rajanit2000) [reply tests/wp-unit/include/Core/Posts/Modules/DeletePostsByRevisionModuleTest.php]: Fixed

File: tests/wp-unit/include/Core/Posts/Modules/DeletePostsByRevisionModuleTest.php [hunk 7]
```diff
@@ -53,26 +100,72 @@ public function test_delete_revisions_for_single_post() {
...
+		$this->assertEquals( 0, count( $trash_posts ) );
+
+		$post = get_post( $post_id );
+		$this->assertEquals( $post->ID, $post_id );
+	}
+
+	/**
+	 * Test trash revisions for a multiple post
+	 */
+	public function test_trash_revisions_for_multiple_post() {
```
Comment (rajanit2000) [inline tests/wp-unit/include/Core/Posts/Modules/DeletePostsByRevisionModuleTest.php]: `test_that_post_revisions_from_multiple_posts_can_be_deleted`

**Pull request merged**